#! /bin/sh -e
## cairo-bug-7514.patch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes debian bug #380064

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
From: Vladimir Vukicevic <vladimir@pobox.com>
Date: Mon, 17 Jul 2006 18:33:47 +0000 (-0700)
Subject: [xlib] Remove XSync implementation of surface_flush
X-Git-Url: http://gitweb.freedesktop.org/?p=cairo;a=commitdiff;h=b7191885c88068dad57d68ced69a752d1162b12c

[xlib] Remove XSync implementation of surface_flush

Remove the xlib implementation of surface_flush which just called XSync
before.
---

--- a/src/cairo-xlib-surface.c
+++ b/src/cairo-xlib-surface.c
@@ -1722,14 +1722,6 @@ _cairo_xlib_surface_get_font_options (vo
     *options = surface->screen_info->font_options;
 }
 
-static cairo_status_t
-_cairo_xlib_surface_flush (void *abstract_surface)
-{
-    cairo_xlib_surface_t *surface = abstract_surface;
-    XSync (surface->dpy, False);
-    return CAIRO_STATUS_SUCCESS;
-}
-
 static void
 _cairo_xlib_surface_scaled_font_fini (cairo_scaled_font_t *scaled_font);
 
@@ -1756,7 +1748,7 @@ static const cairo_surface_backend_t cai
     _cairo_xlib_surface_get_extents,
     NULL, /* old_show_glyphs */
     _cairo_xlib_surface_get_font_options,
-    _cairo_xlib_surface_flush,
+    NULL, /* flush */
     NULL, /* mark_dirty_rectangle */
     _cairo_xlib_surface_scaled_font_fini,
     _cairo_xlib_surface_scaled_glyph_fini,
